src/scenes/Activities/scenes/EventRows.js

```diff
@@ -0,0 +1,103 @@
+import React, { Component, Fragment } from 'react';
+import { inject, observer } from 'mobx-react';
+import { withRouter } from 'react-router';
+import PropTypes from 'prop-types';
+import {
+  TableBody,
+  TableCell,
+  TableRow,
+  withStyles,
+} from '@material-ui/core';
+import cx from 'classnames';
+import { FormattedMessage, injectIntl, intlShape } from 'react-intl';
+
+import styles from './styles';
+import TransactionHistoryID from '../../../components/TransactionHistoryAddressAndID/id';
+import TransactionHistoryAddress from '../../../components/TransactionHistoryAddressAndID/address';
+import { getShortLocalDateTimeString } from '../../../helpers/utility';
+import { i18nToUpperCase } from '../../../helpers/i18nUtil';
+import { getTxTypeString } from '../../../helpers/stringUtil';
+
+const EventRows = observer(({ displayedTxs }) => (
+  <TableBody>
+    {displayedTxs.map((transaction) => (<EventRow key={transaction.txid} transaction={transaction} />))}
+  </TableBody>
+));
+
+@injectIntl
+@withStyles(styles, { withTheme: true })
+@withRouter
+@inject('store')
+@observer
+class EventRow extends Component {
+    static propTypes = {
+      intl: intlShape.isRequired, // eslint-disable-line react/no-typos
+      classes: PropTypes.object.isRequired,
+      transaction: PropTypes.object.isRequired,
+      history: PropTypes.object.isRequired,
+    };
+
+    state = {
+      expanded: false,
+    }
+
+    handleClick = (txid, topicAddress) => async (event) => { // eslint-disable-line
+      if (topicAddress) {
+        const { activities: { history: { getOracleAddress } } } = this.props.store;
+        const { history } = this.props;
+        const nextAddress = await getOracleAddress(topicAddress);
```

Comment:
I don't think we actually need `await` or `async` on this function anymore since we're doing that in the store. Test it out